dascore/proc/filter.py

```diff
@@ -391,7 +391,7 @@ def gaussian_filter(
 def slope_filter(
     patch: PatchType,
     filt: Sequence[float],
-    dims: tuple[str, str] = ("time", "distance"),
+    dims: tuple[str, str] = ("distance", "time"),
```

Comment:
I think we should avoid allowing users to assign this parameter. If we use a different order (dims=("time", "distance")), it will lead to incorrect results.

Reply:
I think we want to leave this in case slopes need to be filtered with coordinates that have different names. This will allow the user to do that. However, I agree it can be a bit confusing, hence the default is what most users will need.

Reply:
I agree we could keep it, but assigning it an incorrect order will lead to inaccurate results. If we choose to keep it, we must adjust the code accordingly. Specifically, when calculating apparent velocity, the numerator and denominator should not be determined by their natural order. Instead, their assignment should be based on the order specified in the `dims` variable.

Reply:
Ya, good point. I think I handled that [here](https://github.com/DASDAE/dascore/blob/8cc1a6f3c5d8da7e18b0005a3750f70d6b142d32/dascore/proc/filter.py#L492-L507) but you are welcome to do additional testing if you like.

Reply:
> Ya, good point. I think I handled that [here](https://github.com/DASDAE/dascore/blob/8cc1a6f3c5d8da7e18b0005a3750f70d6b142d32/dascore/proc/filter.py#L492-L507) but you are welcome to do additional testing if you like.

I've tested it (`dims= ("time","distance",)`) and found that wrong results before I shared the first comment. But you may double check that.

Reply:
> I've tested it (dims= ("time","distance",)) and found that wrong result

Yes, in that case you would be filtering on slowness, as the docstring says the slope is interpreted as the first over the second value, so ("time", "distance") is time/distance. That is why the default  is distance/time (velocity), but I also added support for units so like the `pass_filter` you can indicate slowness or velocity explicitly with units. 

I did some testing in the [fk/slope filter recipe](https://dascore.netlify.app/recipes/fk) and it looked ok. Mind taking a look?

Reply:
Those examples seem perfect. I think instead of letting users specify units, it might be more straightforward to choose between slowness or apparent velocity explicitly. Using the order of `time` and `distance` in the `dims` to determine whether it's a slowness filter or an apparent velocity filter is a bit unconventional. So I suggest that if we want to maintain flexibility while improving clarity, we could replace `dims` with `filt_type='slowness' or 'apparent velocity'`. What do you think?

Reply:
I think support units is important for consistency (eg between `pass_filter` and `slope_filter`. I get the concern of ordering the dims not being the most intuitive way for a standard "time", "distance" patch, but wherever possible we are trying to ensure the patch methods are general enough to apply to any dimensions. 

If you want, you could add a separate method called `velocity_filter` which would just call `slope_filter` but might have a different interface that is more straight-forward for the "time" "distance" case.